### PR TITLE
Feature/con 185/chunk pagination

### PR DESCRIPTION
**Pull request by boajer**
Backported version for https://oat-sa.atlassian.net/browse/CON-313,  ( tree of original PRs here https://github.com/oat-sa/tao-core/pull/2747 )

Deployed and QAed at http://test-depp_pbport.playground.kitchen.it.taocloud.org:46791/

Backport towards versions used at  https://github.com/oat-sa/depp-deploy/tree/frdep03aue

https://github.com/oat-sa/extension-tao-mediamanager/pull/246
https://github.com/oat-sa/extension-tao-item/pull/446

### CODE REVIEW

File: models/classes/media/MediaBrowser.php [hunk 1]
```diff
@@ -1,39 +1,43 @@
...
  */
 interface MediaBrowser {
 
+    public function getDirectories(DirectorySearchQuery $params): array;
+
     /**
-     * @param string $parentLink
-     * @param array $acceptableMime
-     * @param int $depth
+     * @deprecated $this->getDirectories should be used instead
```
Comment (pnal) [inline models/classes/media/MediaBrowser.php:32]: this weird

Comment (boajer) [reply models/classes/media/MediaBrowser.php:32]: What is weird here? Intention was to explain what method should be used instead deprecated one.

**Review by gabrielfs7 — APPROVED** [review]
Looks good! Great job @boajer! 

Let's have a small demo, please.

- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Comment by gabrielfs7** [comment]
Seems not related to your task, but build is broken:

```
+ pwd

+ docker run -u 1003:998 -e EXTENSION=tao -v /srv/data/jenkins/workspace/cg-extension-pr_tao-core_PR-2757/build/artefacts/:/var/www/artefacts/ ci_checks:feature-CON-185-chunk_pagination-f410a33c53b17addefa9b65ec8bbdca57b076077 ci_unit_tests

+ cd /var/www

+ '[[' -d tao/test/unit -a -f artefacts/phpunit.xml ]]

+ cp artefacts/phpunit.xml phpunit.xml

+ vendor/bin/phpunit tao/test/unit -c phpunit.xml --fail-on-risky --coverage-clover artefacts/clover.xml



Fatal error: Uncaught Error: Class 'PHPUnit_Framework_TestCase' not found in /var/www/tao/test/unit/AdaptersTest.php:24

Stack trace:

#0 /var/www/vendor/phpunit/phpunit/src/Util/FileLoader.php(59): include_once()

#1 /var/www/vendor/phpunit/phpunit/src/Util/FileLoader.php(47): PHPUnit\Util\FileLoader::load('/var/www/tao/te...')

#2 /var/www/vendor/phpunit/phpunit/src/Framework/TestSuite.php(358): PHPUnit\Util\FileLoader::checkAndLoad('/var/www/tao/te...')

#3 /var/www/vendor/phpunit/phpunit/src/Framework/TestSuite.php(456): PHPUnit\Framework\TestSuite->addTestFile('/var/www/tao/te...')

#4 /var/www/vendor/phpunit/phpunit/src/Runner/BaseTestRunner.php(94): PHPUnit\Framework\TestSuite->addTestFiles(Array)

#5 /var/www/vendor/phpunit/phpunit/src/TextUI/Command.php(181): PHPUnit\Runner\BaseTestRunner->getTest('tao/test/unit', '', Array)

#6 /var/www/vendor/phpunit/phpunit/src/TextUI/Command.php(163): PHPUnit\TextUI\Command->run(Array, true)

#7 /var/www/vendor/phpunit/phpunit/phpunit(61): PHPUnit\TextUI\Command::main()

#8 {main}

  th in /var/www/tao/test/unit/AdaptersTest.php on line 24

script returned exit code 255
```

File: models/classes/media/MediaBrowser.php (hunk 1 above)
Comment (pnal) [reply models/classes/media/MediaBrowser.php:32]: Sorry I been confused by the names plural/not plural. Everything is right.

**Review by shpran — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Pull request merged**